typescript/sdk/src/hook/HyperlaneHookDeployer.ts

```diff
@@ -42,10 +43,15 @@ export class HyperlaneHookDeployer extends HyperlaneDeployer<
     multiProvider: MultiProvider,
     readonly core: ChainMap<Partial<CoreAddresses>>,
     readonly ismFactory: HyperlaneIsmFactory,
-    readonly igpDeployer = new HyperlaneIgpDeployer(multiProvider),
+    contractVerifier?: ContractVerifier,
+    readonly igpDeployer = new HyperlaneIgpDeployer(
+      multiProvider,
+      contractVerifier,
+    ),
```

Comment:
why reorder here?

Reply:
to reference it when creating the default HyperlaneIgpDeployer (line 49)

Reply:
oh duh

Reply:
We should be using objects for SDK classes to allow us to add params without worrying about order

Reply:
completely agree, I've noticed it used in some places but not others

will try to improve this going forward